app/src/main/java/io/homeassistant/companion/android/widgets/common/WidgetAuthenticationActivity.kt

```diff
@@ -0,0 +1,51 @@
+package io.homeassistant.companion.android.widgets.common
+
+import android.appwidget.AppWidgetManager
+import android.content.Intent
+import android.os.Bundle
+import android.util.Log
+import android.widget.Toast
+import androidx.appcompat.app.AppCompatActivity
+import io.homeassistant.companion.android.authenticator.Authenticator
+import io.homeassistant.companion.android.common.R
+import io.homeassistant.companion.android.widgets.button.ButtonWidget
+
+class WidgetAuthenticationActivity : AppCompatActivity() {
+    companion object {
+        private const val TAG = "WidgetAuthenticationA"
+    }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        Log.d(TAG, "onCreate in WidgetAuthenticationActivity called")
+
+        val authenticator = Authenticator(this, this, ::authenticationResult)
+        authenticator.authenticate(getString(R.string.biometric_set_title))
+    }
+
+    private fun authenticationResult(result: Int) {
+        when (result) {
+            Authenticator.SUCCESS -> {
+                Log.d(TAG, "Authentication successful, calling requested service")
+                val appWidgetId = getIntent().getIntExtra(AppWidgetManager.EXTRA_APPWIDGET_ID, -1)
```

Comment:
Instead of using `getIntent()`, you can simply use `intent` (also suggested by Android Studio).

Reply:
Yeah, I'm new to Kotlin...